code/controlconfig/controlsconfigcommon.cpp

```diff
@@ -1351,25 +1353,16 @@ void control_config_common_read_tbl() {
  * @returns 0 if successful
  * @returns 1 if not successful - nothing was saved
  */
-int control_config_common_write_tbl(bool overwrite = false) {
-	if (cf_exists_full("controlconfigdefaults.tbl", CF_TYPE_TABLES) && !overwrite) {
-		// File exists, and we're told to not overwrite it. Bail
-		return 1;
-	}
-
-	CFILE* cfile = cfopen("controlconfigdefaults.tbl", "w", CFILE_NORMAL, CF_TYPE_TABLES);
-	if (cfile == nullptr) {
-		// Could not open. Bail.
-		return 1;
-	}
+template<class FILETYPE>
+int control_config_common_write_tbl_segment(FILETYPE* cfile, int preset, int (* puts)(const char *, FILETYPE*) ) {
```

Comment:
Double check that the puts callback fits the same convention as the other cfile handlers.